ledger/evaldebugger.go

```diff
@@ -0,0 +1,66 @@
+// Copyright (C) 2019-2022 Algorand, Inc.
+// This file is part of go-algorand
+//
+// go-algorand is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// go-algorand is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with go-algorand.  If not, see <https://www.gnu.org/licenses/>.
+
+package ledger
+
+import (
+	"fmt"
+
+	"github.com/algorand/go-algorand/config"
+	"github.com/algorand/go-algorand/data/basics"
+	"github.com/algorand/go-algorand/data/bookkeeping"
+	"github.com/algorand/go-algorand/data/transactions"
+	"github.com/algorand/go-algorand/ledger/internal"
+	"github.com/algorand/go-algorand/ledger/ledgercore"
+)
+
+// DebuggerLedgerForEval is a ledger interface for the debugger.
+type DebuggerLedgerForEval interface {
+	internal.LedgerForEvaluator
+	Latest() basics.Round
+}
+
+// EvalForDebugger processes a transaction group for the debugger.
+func EvalForDebugger(l DebuggerLedgerForEval, stxns []transactions.SignedTxn) (ledgercore.StateDelta, []transactions.SignedTxnInBlock, error) {
+	prevBlockHdr, err := l.BlockHdr(l.Latest())
```

Comment:
@jdtzmn Is it _safe_ to perform `prevBlockHdr, err := l.BlockHdr(l.Latest())` here?

* For example, it's _possible_ the returned `BlockHeader` here _differs_ from the value applied in `s.checkWellFormed(txgroup)`.
* I'm seeking a double check, but my intuition is:  A single `BlockHeader` should be applied throughout simulation.  
  * If that's accurately - Concretely, `SimulateSignedTxGroup` owns the latest block header side-effect.

Reply:
I believe this is fixed within https://github.com/algorand/go-algorand/pull/4322/commits/6ec12f13380957fa31f7f2a6df6c7721a354a9c8 in which I "patch" the ledger passed to the simulator so that its `Latest()` is pinned.

Reply:
@jdtzmn I agree the issue is fixed via https://github.com/algorand/go-algorand/commit/6ec12f13380957fa31f7f2a6df6c7721a354a9c8, but I have a couple of additional comments.  Both are optional though I feel more strongly about the 1st remark.
* I think there's less chance for error by making `checkWellFormed` and `SimulateSignedTxGroup` accept either (a) the round or (b) the blocker header from which simulation begins.  Conceptually, I think it's _more_ correct for `SimulateSignedTxGroup` to identify the round from which simulation begins independent of `Latest` overrides. 
* I'd lightly consider renaming `simulatorLedger.latest` to something like `simulatorLedger.simulationStart` or similar.  In the context of the simulation ledger, the field represents when simulation starts.  I view it as somewhat incidental that the _latest_ round is used for initialization.

If you'd prefer to keep as is, then I welcome you to close the thread.  Also if you prefer a verbal, happy to do so - let me know.